components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1694,6 +1730,54 @@ private Set<org.wso2.carbon.user.core.common.User> filterUsers(Node node, int of
         }
     }
 
+    /**
+     * Method to get User Count by Group filter
+     *
+     * @param node       Expression or Operation node.
+     * @param domainName Domain name.
+     * @return User count for the filtered group.
+     * @throws CharonException     Error while filtering the users.
+     * @throws BadRequestException Exception occurred due to a bad request.
+     */
+    private int getUserCountByGroup(Node node, String domainName)
+            throws CharonException, BadRequestException {
+
+        int count = 0;
+        // Set filter values.
+        String attributeName = ((ExpressionNode) node).getAttributeValue();
+        String filterOperation = ((ExpressionNode) node).getOperation();
+        String attributeValue = ((ExpressionNode) node).getValue();
+
+        /*
+        If there is a domain and if the domain separator is not found in the attribute value, append the domain
+        with the domain separator in front of the new attribute value.
+        */
+        attributeValue = UserCoreUtil.addDomainToName(((ExpressionNode) node).getValue(), domainName);
```

Comment:
Do we need the extra assignment in 1749?

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/553